### ISSUE TITLE
Dictionary unpacking of a TypedDict yields incorrect `Dict[str, object]` type

### ISSUE THREAD

**Issue by juanpablos** [microsoft/pylance-release#1328]
## Environment data
-   Language Server version: 2021.5.3
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.5

## Snippet

```python
from typing import TypedDict


class Config(TypedDict):
    arg1: int
    arg2: str


def function(arg1: int, arg2: str):
    ...

my_configs: Config = {"arg1": 10, "arg2": "something"}

####
function(**my_configs)
configs_copy = {**my_configs}
values = my_configs.values()
```

## Expected behaviour

- `function(**my_configs)` should pass the type check
- `configs_copy = {**my_configs}` should either have an inferred type of `Dict[str, Union[int, str]]` or `Dict[str, Any]`, or at least `Dict[str, Unknown]`
- `my_configs.values()` should, same as above, have at least `ValuesView[Unknown]` inferred type

## Actual behaviour

- `function(**my_configs)` fails to type check with the following message:
```
(variable) my_configs: Config
Argument of type "object" cannot be assigned to parameter "arg1" of type "int" in function "function"
  "object" is incompatible with "int"PylancereportGeneralTypeIssues
```
- `configs_copy = {**my_configs}` has an inferred type of `Dict[str, object]`
- `my_configs.values()` is inferred as `ValuesView[object]`

## Other info

This type of TypedDict unpacking was working before, though not inferring the types of the keys of the dictionary. However, we can agree that unpacking a TypedDict should not yield an `object` type. 
Work on this bug could end up fulfilling (at least partially) this feature request I did some time ago #374

**Comment by erictraut** [comment]
Thanks for the feature request.

I've added support in the function call logic to support unpacked TypedDict arguments. It now properly handles both allowed and disallowed cases. This will be included in the next release.

I haven't changed the logic for unpacked TypedDict expressions in other contexts because I'm not yet convinced that `Dict[str, object]` is the wrong answer here. This is consistent with the behavior dictated by the "dict" class definition in typeshed, and it's consistent with mypy's behavior.

If you want the unpacked form of your TypedDict to be typed as `Dict[str, Any]`, you can use a type annotation, like this:
```python
configs_copy: Dict[str, Any] = {**my_configs}
```

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.5.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202154-26-may-2021
